src/Injector.hack

```diff
@@ -112,31 +113,18 @@ abstract class HTMLPurifier_Injector {
 	 */
 	public function checkNeeded(HTMLPurifier_Config $config): string {
 		$def = TypeAssert\matches<Definition\HTMLPurifier_HTMLDefinition>($config->getHTMLDefinition());
-		$i = 0;
-		$index = 0;
 		foreach ($this->needed as $element => $attributes) {
-			if ($i is int) {
-				$i = $element;
+			if (!HH\idx($def->info, $element)) {
```

Comment:
Would `C\contains_key` work here?

If so, I think that would be more clear here and on 124, since the `HH\idx` returns a default if it's not present and the default is NULL so it would effectively resolve to `if (!NULL)` which isn't ideal.